### DISCUSSION TITLE
Mistral AI support for Spring AI

### DISCUSSION THREAD

**Discussion by ricken07** [spring-projects/spring-ai#366] [Ideas]
Hi, 
for a while I've taken the initiative to work on the implementation of MistralAI support by SpringAI on the 'mistral-ai' branch you can see the progress. 
@markpollack @tzolov can you take a tour to see

**Comment by tzolov** [comment]
Now we have full Mistral AI support (@ricken07 thanks for the contributions!)
You can more in the reference docs:
 - [Mistral AI Chat Completion](https://docs.spring.io/spring-ai/reference/0.8-SNAPSHOT/api/clients/mistralai-chat.html) - sync and streaming
 - [Mistral AI Embedding](https://docs.spring.io/spring-ai/reference/0.8-SNAPSHOT/api/embeddings/mistralai-embeddings.html)
 - [Mistral AI Function Calling](https://docs.spring.io/spring-ai/reference/0.8-SNAPSHOT/api/clients/functions/mistralai-chat-functions.html) - Note that currently Mistral AI does not support parallel function calling!
